### DISCUSSION TITLE
[schema generator] Where is the schema returning from toSchema() supposed to go? (Spring GraphQL)

### DISCUSSION THREAD

**Discussion by ivargj** [ExpediaGroup/graphql-kotlin#1597] [General] [answered]
Hi! I am trying to use the schema generator to replace manually maintaining the schema file in our existing Spring Boot GraphQL module. 
Looking at the [generator documentation](https://opensource.expediagroup.com/graphql-kotlin/docs/schema-generator/schema-generator-getting-started), it describes how to use the `toSchema()` function to generate the schema, but I can't find any documentation or [example code](https://github.com/ExpediaGroup/graphql-kotlin/tree/master/examples) on how to configure my spring graphql application to use that schema. 

I have tested the schema generation itself through debugging the code, and it seems to succeed in generating the schema. I just don't know how to get the GraphQL instance to pick it up.

One of my thoughts was to use the [GraphQLSource builder](https://docs.gitcode.net/spring/guide/en/spring-for-graphql/spring-graphql.html#_4-1-graphqlsource) to customize the schema. But as of now I only get a MissingSchemaException when trying that. 

I could probably solve this by rebuilding a graphql server based on the graphql-kotlin-spring-server here, but as I only need the schema generation functionality, I was hoping it was possible to simply use my existing code with the schema generator on top. 

Appreciate any help!

**Comment by dariuszkuc** [comment, marked as answer]
Hello 👋 
`spring-graphql` is a completely separate project that uses SDL first approach to load schema file and automatically map resolvers based on some annotations (e.g. `@QueryMapping`/`@SchemaMapping`). `graphql-kotlin` follows a code first approach where code is used to define the schema which is constructed dynamically at runtime.

While you could potentially wire up `graphql-kotlin` generated schema through `GraphQlSourceBuilderCustomizer` and `schemaFactory` method (similar to https://github.com/spring-projects/spring-graphql/issues/384) but I don't think thats a good idea - basically you would loose all the runtime wiring information provided by `spring-graphql` so all your Spring annotated methods would stop working (or if you somehow merge those two wiring they could possibly conflict with wiring generated by `graphql-kotlin`).

**TLDR it is potentially doable but most likely causes more issues than benefits.** Our `graphql-kotlin-spring-server` is a SpringBoot server so migration should be fairly simple. If there is some missing functionality there, please raise an issue.

**Comment by samuelAndalon** [comment]
as @dariuszkuc mentioned, technically it could be possible to reload the `GraphQL` instance from `graphql-java` as it allows the instance to mutate it with the `transform` method, however you will be loosing all the `GraphQLSchema` run timewiring of fields and resolvers.

**Comment by ivargj** [comment]
Thanks for your clearifying answers, guys! I'll have a consideration on whether to migrate the whole thing or not. 

Out of curiosity, is there really a use case for usuing the schema generator on its own? Or does it only make sense to use as a part of graphql-kotlin-spring-server?
